gluonnlp/embedding/evaluation.py

```diff
@@ -429,43 +482,34 @@ def __init__(self, idx_to_vec, analogy_function='ThreeCosMul', k=1,
         self.k = k
         self.exclude_question_words = exclude_question_words
 
-        self._internal_k = self.k + 3 * self.exclude_question_words
-
         with self.name_scope():
-            self.analogy = create(kind='analogy', name=analogy_function,
-                                  idx_to_vec=idx_to_vec, k=self._internal_k)
+            self.analogy = create(
+                kind='analogy',
+                name=analogy_function,
+                idx_to_vec=idx_to_vec,
+                k=self.k,
+                exclude_question_words=exclude_question_words)
 
         if not isinstance(self.analogy, WordEmbeddingAnalogyFunction):
             raise RuntimeError(
                 '{} is not a WordEmbeddingAnalogyFunction'.format(
                     self.analogy.__class__.__name__))
 
-    def forward(self, words1, words2, words3):  # pylint: disable=arguments-differ
-        """Implement forward computation.
+    def hybrid_forward(self, F, words1, words2, words3):  # pylint: disable=arguments-differ, unused-argument
+        """Compute analogies for given question words.
 
         Parameters
         ----------
-        words1 : NDArray
-            Word indices.
-        words2 : NDArray
-            Word indices.
-        words3 : NDArray
-            Word indices.
+        words1 : Symbol or NDArray
+            Word indices of first question words. Shape (batch_size, ).
+        words2 : Symbol or NDArray
+            Word indices of second question words. Shape (batch_size, ).
+        words3 : Symbol or NDArray
+            Word indices of third question words. Shape (batch_size, ).
 
         Returns
         -------
-        predicted_indices : NDArray
-            Predicted indices of shape (batch_size, k)
+        predicted_indices : Symbol or NDArray
+            Indices of predicted analogies of shape (batch_size, k)
         """
-        pred_idxs = self.analogy(words1, words2, words3)
```

Comment:
Are these changes caused by transform refactoring, or it's not related?

Reply:
Only related in so far as I updated the `word_embedding_training.ipynb` due to the changes of this PR, looked at the underlying code and decided to remove this bottleneck.